### ISSUE TITLE
neo4j build failure!

### ISSUE THREAD

**Issue by 1261972163** [neo4j/neo4j#12300]
Ubuntu 16.04.4 LTS
root@LAPTOP-J9FOAR91:~/neo4j# mvn -v
Apache Maven 3.6.2 (40f52333136460af0dc0d7232c0dc0bcf0d9e117; 2019-08-27T23:06:16+08:00)
Maven home: /opt/maven/apache-maven-3.6.2
Java version: 1.8.0_171, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-oracle/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.4.0-18362-microsoft", arch: "amd64", family: "unix"

git clone https://github.com/neo4j/neo4j.git
cd neo4j
mvn clean install 

root@LAPTOP-J9FOAR91:~/neo4j# mvn clean install
......
......
......
Downloaded from nexus-aliyun: http://maven.aliyun.com/nexus/content/groups/public/org/apache/commons/commons-compress/1.16.1/commons-compress-1.16.1.jar (560 kB at 320 kB/s)
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Neo4j 3.5.11-SNAPSHOT:
[INFO]
[INFO] Neo4j - Build resources ............................ SUCCESS [  2.229 s]
[INFO] Neo4j .............................................. SUCCESS [ 15.131 s]
[INFO] Neo4j - Annotation Processors ...................... SUCCESS [  4.892 s]
[INFO] Zstandard .......................................... SUCCESS [  0.955 s]
[INFO] Neo4j - Common ..................................... SUCCESS [  9.655 s]
[INFO] Neo4j - Resource interface ......................... SUCCESS [  3.080 s]
[INFO] Neo4j - Unsafe Access .............................. SUCCESS [  1.747 s]
[INFO] Neo4j - Collections ................................ SUCCESS [  8.167 s]
[INFO] Neo4j - Graph Database API ......................... SUCCESS [  7.995 s]
[INFO] Neo4j - Concurrent ................................. SUCCESS [  2.152 s]
[INFO] Neo4j - IO ......................................... SUCCESS [ 11.341 s]
[INFO] Neo4j - Logging .................................... SUCCESS [ 41.334 s]
[INFO] Neo4j - Configuration .............................. SUCCESS [  1.709 s]
[INFO] Neo4j - Procedure API .............................. FAILURE [30:09 min]
[INFO] Neo4j - Diagnostics ................................ SKIPPED
[INFO] Neo4j - Value ...................................... SKIPPED
[INFO] Neo4j - Random Value ............................... SKIPPED
[INFO] Neo4j - Kernel API ................................. SKIPPED
[INFO] Neo4j - CSV reading and parsing .................... SKIPPED
[INFO] Neo4j - Lucene Index Upgrade ....................... SKIPPED
[INFO] Neo4j - Native index ............................... SKIPPED
[INFO] Neo4j - Native Spatial Index ....................... SKIPPED
[INFO] Neo4j - SSL ........................................ SKIPPED
[INFO] Neo4j - Graph Database Kernel ...................... SKIPPED
[INFO] Neo4j - Command Line ............................... SKIPPED
[INFO] Neo4j - Lucene Index ............................... SKIPPED
[INFO] Neo4j - Import Command Line Tool ................... SKIPPED
[INFO] Neo4j - JMX support ................................ SKIPPED
[INFO] Neo4j - Database Management System ................. SKIPPED
[INFO] Neo4j - Graph Algorithms ........................... SKIPPED
[INFO] Neo4j - Code Generator ............................. SKIPPED
[INFO] Neo4j - Legacy ..................................... SKIPPED
[INFO] Neo4j - Community Cypher Build ..................... SKIPPED
[INFO] openCypher Front End Parent ........................ SKIPPED
[INFO] openCypher Utils ................................... SKIPPED
[INFO] openCypher Expressions ............................. SKIPPED
[INFO] openCypher AST for the Cypher Query Language ....... SKIPPED
[INFO] openCypher Parser .................................. SKIPPED
[INFO] openCypher Rewriting ............................... SKIPPED
[INFO] openCypher Front End ............................... SKIPPED
[INFO] Neo4j - Cypher Intermediate Representation ......... SKIPPED
[INFO] Neo4j - Cypher Logical Plans ....................... SKIPPED
[INFO] Neo4j - Cypher Planner SPI ......................... SKIPPED
[INFO] Neo4j - Cypher Planner ............................. SKIPPED
[INFO] Neo4j - Cypher Runtime Utilities ................... SKIPPED
[INFO] Neo4j - Cypher Interpreted Runtime ................. SKIPPED
[INFO] Neo4j - Cypher ..................................... SKIPPED
[INFO] Neo4j - Security ................................... SKIPPED
[INFO] Neo4j - SLF4J Neo4j Binding ........................ SKIPPED
[INFO] Neo4j - Usage Data Collection ...................... SKIPPED
[INFO] Neo4j - Fulltext index ............................. SKIPPED
[INFO] Neo4j - Data Collector ............................. SKIPPED
[INFO] Neo4j - Bolt Protocol Version 1 .................... SKIPPED
[INFO] Neo4j - Consistency Checker ........................ SKIPPED
[INFO] Neo4j - Community .................................. SKIPPED
[INFO] Neo4j - Community (Bill of Materials) .............. SKIPPED
[INFO] Neo4j - Server API ................................. SKIPPED
[INFO] Neo4j - Community Integration Tests ................ SKIPPED
[INFO] Neo4j - Community Integration Test Support ......... SKIPPED
[INFO] Neo4j - Server ..................................... SKIPPED
[INFO] Neo4j - Server Plugin Tests ........................ SKIPPED
[INFO] Neo4j - Test Harness ............................... SKIPPED
[INFO] Neo4j - Procedure Compiler Core .................... SKIPPED
[INFO] Neo4j - Procedure Compiler Parent POM .............. SKIPPED
[INFO] Neo4j - Procedure Compiler Integration Tests ....... SKIPPED
[INFO] Neo4j - Community Kernel Integration Tests ......... SKIPPED
[INFO] Neo4j - Community Index Integration Tests .......... SKIPPED
[INFO] Neo4j - Community Import Integration Tests ......... SKIPPED
[INFO] Neo4j - Community DBMS Integration Tests ........... SKIPPED
[INFO] Neo4j - Community Graph Algorithms Integration Tests SKIPPED
[INFO] Neo4j - Community Consistency Checker Integration Tests SKIPPED
[INFO] Neo4j - Community Bolt Integration Tests ........... SKIPPED
[INFO] Neo4j - Community Generic Integration Tests ........ SKIPPED
[INFO] Neo4j - Community Cypher Integration Tests ......... SKIPPED
[INFO] Neo4j - Push To Cloud - Admin command .............. SKIPPED
[INFO] Neo4j - Cypher Specification Suite Tools ........... SKIPPED
[INFO] Neo4j - Cypher Acceptance .......................... SKIPPED
[INFO] Neo4j - Cypher Compatibility Specification Suite ... SKIPPED
[INFO] Neo4j - Cypher Expression Evaluator ................ SKIPPED
[INFO] Neo4j - Community Build ............................ SKIPPED
[INFO] Neo4j - Server Assembler ........................... SKIPPED
[INFO] Neo4j - Community Server Assembler ................. SKIPPED
[INFO] Neo4j - Packaging Build ............................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  32:01 min
[INFO] Finished at: 2019-09-22T16:58:49+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.0.1:javadoc-no-fork (generate-javadocs) on project neo4j-procedure-api: Execution generate-javadocs of goal org.apache.maven.plugins:maven-javadoc-plugin:3.0.1:javadoc-no-fork failed: Plugin org.apache.maven.plugins:maven-javadoc-plugin:3.0.1 or one of its dependencies could not be resolved: Could not transfer artifact org.apache.httpcomponents:httpcore:jar:4.4.4 from/to nexus-aliyun (http://maven.aliyun.com/nexus/content/groups/public): Read timed out -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.

**Comment by chrisvest** [comment]
You have a weird Maven repository configuration somewhere (aliyun.com), and Maven failed to download a plug-in dependency. Related S/O question: https://stackoverflow.com/questions/28725195/maven-clean-install-fails-with-connection-timed-outconnect-error

**Comment by 1261972163** [comment]
Today I modified, continued to build, and failed again. As follows:
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  05:27 min
[INFO] Finished at: 2019-10-03T11:00:39+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.revapi:revapi-maven-plugin:0.10.5:check (default) on project neo4j-jmx: The following API problems caused the build to fail:
[ERROR] java.class.added: @interface org.neo4j.jmx.Description: Class was added.
[ERROR] java.class.added: class org.neo4j.jmx.JmxUtils: Class was added.
[ERROR] java.class.added: interface org.neo4j.jmx.Kernel: Class was added.
[ERROR] java.class.added: @interface org.neo4j.jmx.ManagementInterface: Class was added.
[ERROR] java.class.added: interface org.neo4j.jmx.Primitives: Class was added.
[ERROR] java.class.added: interface org.neo4j.jmx.StoreFile: Class was added.
[ERROR] java.class.added: interface org.neo4j.jmx.StoreSize: Class was added.
[ERROR]
[ERROR] Consult the plugin output above for suggestions on how to ignore the found problems.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
